### ISSUE TITLE
i3ipc-glib no longer needed

### ISSUE THREAD

**Issue by ghost** [Lokaltog/candybar#96]
Now that EWMH support is in the latest i3 stable release would it be possible to remove this as an optional dependency (both from the project and the pkgbuild)? I don't know if i3's EWMH implementation is feature complete compared to i3ipc-glib, I assume @acrisci would know.

**Comment by acrisci** [comment]
I was not able to complete my implementation of every required EWMH hint because it turned out to be more difficult than I thought and I got busy with some other things.

I will be working on implementing the required hints in the next version of i3. When these are done, the present implementation of the desktop widget will be the same as the EWMH version (as far as I know).

However, the i3 ipc protocol goes above and beyond EWMH hints, which were never designed for tiling window managers, so a creative person could still use this library to design new features for Candybar.

tl;dr: for now, yes but probably not for long.

**Comment by acrisci** [comment]
The i3 development branch now supports the desktop names and indication of the current desktop. Desktop switching does not work, but I don't think it is implemented in Candybar yet.

**Comment by ghost** [comment]
By desktop switching do you mean workspaces? I doubt you do, because IIRC you wrote the desktops_i3 plugin.

**Comment by acrisci** [comment]
Actually, I did not write the desktops_i3 plugin, but I did write the i3ipc-glib library.

We interpret the idea of a EWMH "desktop" as an i3 "workspace" which is an ersatz replacement, but this is the spec we have.

When I click on a workspace button, I expect candybar to give me a _NET_CURRENT_DESKTOP request to change the desktop. That request will be handled by i3 soon.

So it seems to me that the development version of i3 supports the desktops extension and there is no need for the desktops_i3 extension unless someone has plans to expand it to use the features of the i3 command interface that cannot be supported by EWMH hints and requests.

I would wait until the next release of i3 to consider removing desktops_i3.

**Comment by ghost** [comment]
Allright, thanks for clearing that up. I forgot it was @5paceManSpiff who made said plugin.

**Comment by Lokaltog** [comment]
I'm going to at least wait until the next release of i3 before removing the desktop widget. The `_NET_CURRENT_DESKTOP` requests will be handled by JSCore later, similar to how the play controls work on the mpris widget.
